src/components/general/TimePicker.js

```diff
@@ -30,10 +28,33 @@ class TimePicker extends Component {
 		};
 	}
 
-	static propTypes = {
-		value: PropTypes.string,
-		placeholder: PropTypes.string.isRequired,
-		onSelect: PropTypes.func.isRequired
+	componentDidUpdate(prevProps) {
+		if (prevProps.value !== this.props.value) {
+			let [time, isInitialized] = this.initializeTimePicker();
+
+			this.setState({
+				hour: isInitialized ? time[0] : "",
+				minute: isInitialized ? time[1] : "",
+				period: isInitialized ? time[2] : "",
+				focused: isInitialized
+			});
+		}
+	}
+
+	initializeTimePicker() {
+		let time = [];
+
+		if (this.props.value) {
+			time = this.props.value.split(":");
```

Comment:
this should destructure the array that comes out of the split to properly name the variables at each section of the array.
```suggestion
			const [hour,minute,period] = this.props.value.split(":");
```
or whatever it would be. Make appropriate changes to however you use time below.